### PR TITLE
Add visual studio project files

### PR DESCRIPTION
**Pull request by Brett208**
~~I get the following error:~~

~~Error (active)	E2512	the argument to a feature-test macro must be a simple identifier	JsonMap	\JsonMap\packages\nlohmann.json.3.7.0\build\native\include\nlohmann\json.hpp	1274~~

~~However, it also says build completed. So I don't know if this would stop JsonMap from being usable by a consuming executable or dll. Since I made it a static library, it is hard to test this theory out of the box.~~ EDIT: Error does not exist with the updated location of the Project File. It may have been related to differing versions of the JSON library, but not sure.

Also, had to add preprocessor conditional logic because the VS Nuget package wanted to include a subfolder on the json header. Maybe a better way to manage with project settings?

### CODE REVIEW

**Comment by Brett208** [comment]
I cannot seem to get the packages.config file to register if the project file is in the src subdirectory. Not sure what I'm doing wrong though...

**Comment by DanRStevens** [comment]
Err, might be a slight misunderstanding here. The JsonMap project should produce an executable. It uses static library inputs, but the project itself is an executable.

File: src/mapToJson.h [hunk 1]
```diff
@@ -1,6 +1,11 @@
 #include "OP2Utility.h"
+#if _WIN32
+#include <nlohmann/json.hpp>
+#else
 #include <json.hpp>
+#endif
```
Comment (DanRStevens) [inline src/mapToJson.h:6]: Ahh yes, looks like that's the natural include path for the NuGet package. This should probably be fixed on the Linux side by adjusting the include folder option. Looks like I should strip off the "nlohmann/" part of the `-I` option:
```
CPPFLAGS := -I $(UTILITYDIR)include -I packages/nlohmann.json.3.2.0/build/native/include/
```

That should allow it to build on both platforms without the `#if` include guard.

**Review by DanRStevens — COMMENTED** [review]
The `packages.config` file can be moved into the `src/` folder if needed. Or the project file could be moved to the project root. It's a little unclear to me what is affected there.

**Comment by Brett208** [comment]
Might be worse squashing some of the commits before merging to reduce commit noise if you are okay with rewriting history. Any reason to be using version 3.2 of nlomann JSON instead of the current 3.7 version?

**Comment by DanRStevens** [comment]
No reason for the older version of nlohman JSON. That's likely just what was out when I added the dependency about ~~half~~ a year ago.

Squashing/rewriting history should be fine. I created PR #7 for the include path change listed in the comments above. Feel free to merge it and rebase on top of it if you want. It might help cleanup the header include change you needed to make here.

**Comment by Brett208** [comment]
See PR #8 for manual rebase of this work.

**Pull request closed without merging**